code/game/objects/items/reagent_containers/food/snacks.dm

```diff
@@ -2792,3 +2792,72 @@
 		if("spiced apples", "chocolate brownie", "sugar cookie", "choco bar", "crayon")
 			icon_state = "dessert"
 			list_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/sugar = 1)
+
+
+/obj/item/reagent_containers/food/snacks/lollipop
+	name = "lollipop"
+	desc = "A delicious lollipop."
+	icon = 'icons/obj/items/lollipop.dmi'
+	icon_state = "lollipop_stick"
+	item_state = "lollipop_stick"
+	flags_equip_slot = ITEM_SLOT_MASK
+	list_reagents = list(/datum/reagent/consumable/nutriment = 5, /datum/reagent/iron = 10, /datum/reagent/consumable/sugar = 10)
+	var/mutable_appearance/head
+	var/headcolor = rgb(0, 0, 0)
+	var/succ_dur = 180
+	var/succ_int = 100
+	var/next_succ = 0
+	tastes = list("candy" = 1)
+ 
+/obj/item/reagent_containers/food/snacks/lollipop/Initialize()
+	. = ..()
+	head = mutable_appearance('icons/obj/items/lollipop.dmi', "lollipop_head")
+	change_head_color(rgb(rand(0, 255), rand(0, 255), rand(0, 255)))
+ 
+//makes lollipops actually wearable as masks and still edible the old fashioned way.
+/obj/item/reagent_containers/food/snacks/lollipop/proc/handle_reagents()
+		if(iscarbon(loc))
```

Comment:
Given the check in the other proc this should necessarily be true and need not be checked.
You can pass the carbon mob by reference doing something like calling `handle_reagents(loc)` and adding that variable to the proc: `handle_reagents(mob/living/carbon/user)`.

Reply:
I am stupid and require further explanation